src/modules/r.rs

```diff
@@ -0,0 +1,101 @@
+use regex::Regex;
+
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::r::RConfig;
+use crate::formatter::StringFormatter;
+
+const R_VERSION_PATTERN: &str = r" (?P<rversion>\d+\.\d+\.\d+) ";
+
+/// Creates a module with the current R programming language version.
+///
+/// Will display the R programming language version if any of the following criteria are met:
+///     - Current directory contains a `.R` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("r");
+    let config: RConfig = RConfig::try_load(module.config);
+
+    let is_r_project = context
+        .try_begin_scan()?
+        .set_extensions(&["R", "Rproj"])
+        .is_match();
+    if !is_r_project {
+        return None;
+    }
+
+    let r_version = context.exec_cmd("R", &["--version"])?.stderr;
+    let formatted_version = parse_version(&r_version)?;
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => Some(Ok(&formatted_version)),
+                _ => None,
+            })
+            .parse(None)
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `r`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn parse_version(version: &str) -> Option<String> {
+    let version_regex = Regex::new(R_VERSION_PATTERN).ok()?;
+    let captures = version_regex.captures(version)?;
+    let r_version = captures["rversion"].to_owned();
+    let r_formatted = format!("v{}", r_version);
+    Some(r_formatted.trim().to_owned())
+}
```

Comment:
This can be done without using regex:
```suggestion
fn parse_version(r_version: &str) -> Option<String> {
    let version = r_version
        .lines()
        // take first line
        .next()
        // split into ["R", "version", "3.6.3", "(2020-02-29)", ...]
        .split_whitespace()
        // return "3.6.3"
        .nth(2)?;

    Some(format!("v{}", r_version))
}
```

Reply:
I had to modify this code a bit, if you want to take a second look.